bokehjs/src/lib/models/annotations/slope.ts

```diff
@@ -29,17 +29,29 @@ export class SlopeView extends AnnotationView {
     const xscale = this.coordinates.x_scale
     const yscale = this.coordinates.y_scale
 
-    const sy_start = frame.bbox.top
-    const sy_end = sy_start + frame.bbox.height
+    let sy_start
+    let sy_end
+    let sx_start
+    let sx_end
+    if (gradient == 0) {
+      sy_start = yscale.compute(y_intercept)
+      sy_end = sy_start
 
-    const y_start = yscale.invert(sy_start)
-    const y_end = yscale.invert(sy_end)
+      sx_start = frame.bbox.left
+      sx_end = sx_start + frame.bbox.width
+    } else {
+      sy_start = frame.bbox.top
+      sy_end = sy_start + frame.bbox.height
 
-    const x_start = (y_start - y_intercept) / gradient
-    const x_end = (y_end - y_intercept) / gradient
+      const y_start = yscale.invert(sy_start)
+      const y_end = yscale.invert(sy_end)
 
-    const sx_start = xscale.compute(x_start)
-    const sx_end = xscale.compute(x_end)
+      const x_start = (y_start - y_intercept) / gradient
+      const x_end = (y_end - y_intercept) / gradient
+
+      sx_start = xscale.compute(x_start)
+      sx_end = xscale.compute(x_end)
+    }
```

Comment:
This method is getting a bit logn. Would suggest pulling the `sx_start`, `sy_end`, etc code in to a helper method. Then you can also avoid the `let` statements by using early return:
```
if (gradient == 0) {
  // compute zero case
  return [...]
}
// compute non-zero case
return [...]
```

Reply:
@bryevdv, thanks for your suggestion. I tried a few things, but I lack the fluency in typescript to get the scoping right without VSCode warnings and without using a lot of code. Any pointer is welcome. I now reduced it by three lines by declaring the variables in one as I saw also done in the arrow.ts in the same directory.